lib/deploy/data.ts

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    GitProject,
+    HandlerContext,
+    logger,
+    Project,
+} from "@atomist/automation-client";
+import {
+    GoalInvocation,
+    SdmGoalEvent,
+} from "@atomist/sdm";
+import { readSdmVersion } from "@atomist/sdm-core";
+import * as k8s from "@kubernetes/client-node";
+import * as dockerfileParser from "docker-file-parser";
+import * as stringify from "json-stringify-safe";
+import * as _ from "lodash";
+import { DeepPartial } from "ts-essentials";
+import { defaultNamespace } from "../kubernetes/namespace";
+import { KubernetesApplication } from "../kubernetes/request";
+import {
+    goalEventSlug,
+    KubernetesDeploy,
+    KubernetesDeployRegistration,
+} from "./goal";
+import { loadKubernetesSpec } from "./spec";
+
+/**
+ * JSON propery under the goal event data where the
+ * [[KubernetesDeployment]] goal [[KubernetesApplication]] data are
+ * stored, i.e., the value of `goal.data[sdmPackK8s]`.
+ */
+export const sdmPackK8s = "sdm-pack-k8s";
```

Comment:
I started to namespace those keys, eg. in https://github.com/atomist/sdm/blob/master/lib/api/registration/ServiceRegistration.ts#L23

Probably should change this to `atomist:sdm/pack-k8s` or similar to follow the convention in sdm.

Reply:
Shouldn't it be `atomist.com/sdm-pack-k8s`?

Reply:
Or `@atomist/sdm-pack-k8s`?

Reply:
`.` is not a good idea as that messes up `_.get`: maybe `@atomist/xxx/xxx` could work. Not sure if we need the @. I really just want to do something consistent.

Reply:
I've changed it to just use the package name, `@atomist/sdm-pack-k8s`. I can drop the "@" if you prefer.

Reply:
Also, I'm not sure why we would have a slash between "sdm" and "pack" and not just stick with a dash.